internal/fs/vss_windows.go

```diff
@@ -697,9 +697,20 @@ func initializeVssCOMInterface() (*ole.IUnknown, uintptr, error) {
 
 	var oleIUnknown *ole.IUnknown
 	result, _, _ := vssInstance.Call(uintptr(unsafe.Pointer(&oleIUnknown)))
+	hresult := HRESULT(result)
```

Comment:
Please add here
```
if hresult != S_OK &&  oleIUnknown != nil {
    oleIUnknown.Release()
}
```
Callers don't need to release in case of errors.

Reply:
I'd like to keep it like this because they need co release on success anyway. Adding a special case will just make the code more complex.

Reply:
The current setup makes the code complex. Idiomatic Go is, e.g.,

```go
f, err := os.Open(filename)
if err != nil {
	return err
}
defer f.Close()
```

with the understanding that the caller should not use f when err == nil, but isn't responsible for disposing of it either. os.Open is responsible for making that possible.